src/target/stm32mp15.c

```diff
@@ -0,0 +1,154 @@
+/*
+ * This file is part of the Black Magic Debug project.
+ *
+ * Copyright (C) 2023 1BitSquared <info@1bitsquared.com>
+ * Written by ALTracer <tolstov_den@mail.ru>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * This file implements STM32MP15 target specific functions for detecting
+ * the device and providing the XML memory map.
+ *
+ * References:
+ * RM0436 - STM32MP157 advanced Arm®-based 32-bit MPUs, Rev. 5
+ *   https://www.st.com/resource/en/reference_manual/rm0436-stm32mp157-advanced-armbased-32bit-mpus-stmicroelectronics.pdf
+ */
+
+#include "general.h"
+#include "target.h"
+#include "target_internal.h"
+#include "cortexm.h"
+
+/* Access from processor address space.
+ * Access via the debug APB is at 0xe0081000 over AP1. */
+#define DBGMCU_IDCODE      0x50081000U
+#define STM32MP15_UID_BASE 0x5c005234U
+
+#define DBGMCU_IDC (DBGMCU_IDCODE + 0U)
+#define DBGMCU_CR  (DBGMCU_IDCODE + 4U)
+#define DBGSLEEP   (1U << 0U)
+#define DBGSTOP    (1U << 1U)
+#define DBGSTBY    (1U << 2U)
+
+#define ID_STM32MP15x        0x5000U /* RM0436 */
```

Comment:
Please prefix this with a comment stating what § and pg this comes from in the RM like we did in the STM32H5 support:
https://github.com/blackmagic-debug/blackmagic/blob/640e853d2478c72a09514b4513da39e1beec2b46/src/target/stm32h5.c#L108-L109

Reply:
Done for both of the observed ID values, referring to `TARGETID==IDCODE` match and `CM4ROM_PIDR4` errata.